src/serlo-editor-repo/plugin-text/components/text-editor.tsx

```diff
@@ -59,19 +59,15 @@ export type TextEditorProps = EditorPluginProps<
 >
 
 export function TextEditor(props: TextEditorProps) {
-  const [hasSelectionChanged, setHasSelectionChanged] = useState(0)
-  const [isLinkNewlyCreated, setIsLinkNewlyCreated] = useState(false)
```

Comment:
Quick explanation of this workaround:

When updating Slate, the idea was to keep the editor functioning as it was before the Slate update. Before the update, the input inside the link controls overlay would only be focused when the link was newly created. If you click on an already existing link, that input would not be focused. Without the workaround, that input will be focused whenever a user clicks on a link element, or navigates to it using the keyboard.

While I hate this workaround and would love to see it go, I have to ask, are you sure you want to introduce this change? 😄

Reply:
as discussed synchronously: I simplified this to rely on the `value`. If a link does not have a `href` set it is considered a new link and the input is focused. 
Let's see what testing shows, but I feel like this is as good or even better since we want people to either set a `href` or remove the link.